reactive/ovn_chassis_charm_handlers.py

```diff
@@ -181,3 +181,54 @@ def provide_chassis_certificates_to_principal():
                             level=ch_core.hookenv.WARNING)
 
     reactive.clear_flag('ovn.certs.changed')
+
+
+@reactive.when_any('config.changed.ovs-exporter-channel',
+                   'snap.installed.prometheus-ovs-exporter')
+def reassess_exporter():
+    is_installed = snap.is_installed('prometheus-ovs-exporter')
+    channel = None
+    with charm.provide_charm_instance() as instance:
+        channel = instance.options.ovs_exporter_snap_channel
+
+    if channel is None:
+        # Attempt to remove the snap if it is present, the snap command
+        # returns 0 if the snap is not installed.
+        snap.remove('prometheus-ovs-exporter')
+        return
+
+    if is_installed:
+        snap.refresh('prometheus-ovs-exporter', channel=channel,
+                     devmode=True)
+    else:
+        snap.install('prometheus-ovs-exporter', channel=channel,
+                     devmode=True)
+
+
+@reactive.when('metrics-endpoint.available',
+               'snap.installed.prometheus-ovs-exporter')
+def handle_metrics_endpoint():
+    metrics_endpoint = reactive.endpoint_from_flag(
+        'metrics-endpoint.available')
+    job_name = 'ovs-exporter'
+    metrics_endpoint.expose_job(
+        job_name,
+        static_configs=[{"targets": ["*:9475"]}])
+
+
+@reactive.when('metrics-endpoint.available')
```

Comment:
The reactive handlers in this layer are opt-in, and all of them should be guarded with:
```python
reactive.when(OVN_CHASSIS_ENABLE_HANDLERS_FLAG, ...
```
You probably also want to assert that the charm is installed by looking for the `charm.installed` flag?

We also do not want to perform any work when in update-status hook, so all handlers should have:
```python
@reactive.when_none('is-update-status-hook', ...
```

Reply:
Addressed in:

https://github.com/openstack-charmers/charm-layer-ovn/pull/74/commits/0df1407c7afabd9c339e963890374696c4205e53
https://github.com/openstack-charmers/charm-layer-ovn/pull/74/commits/ea198cc28b87d8ac6bcc0830a44d1422f7389bc3